unittests/SourceKit/SwiftLang/CursorInfoTest.cpp

```diff
@@ -455,3 +459,40 @@ TEST_F(CursorInfoTest, CursorInfoCancelsPreviousRequest) {
   if (expired)
     llvm::report_fatal_error("Did not receive a resonse for the first request");
 }
+
+TEST_F(CursorInfoTest, CursorInfoCancellation) {
+  // TODO: This test case relies on the following snippet being slow to type
+  // check so that the first cursor info request takes longer to execute than it
+  // takes time to schedule the second request. If that is fixed, we need to
+  // find a new way to cause slow type checking. rdar://80582770
+  const char *SlowDocName = "slow.swift";
+  const char *SlowContents = "func foo(x: Invalid1, y: Invalid2) {\n"
+                             "    x / y / x / y / x / y / x / y\n"
+                             "}\n";
+  auto SlowOffset = findOffset("x", SlowContents);
+  const char *Args[] = {"-parse-as-library"};
+  std::vector<const char *> ArgsForSlow = llvm::makeArrayRef(Args).vec();
+  ArgsForSlow.push_back(SlowDocName);
+
+  open(SlowDocName, SlowContents, llvm::makeArrayRef(Args));
+
+  SourceKitCancellationToken CancellationToken = &CancellationToken;
```

Comment:
A stack address is not necessarily unique in case we add more cancellation tests this could be problematic. Creating a global counter in this test file should be fine.